crawler/s3/tests/unit/test_s3_crawler.py

```diff
@@ -0,0 +1,101 @@
+from datetime import datetime, timedelta, UTC
+from pathlib import Path
+from typing import Any
+
+import botocore.paginate
+
+from crawler.s3.crawler import s3_crawler
+from crawler.s3.crawler.s3_crawler import S3Crawler
+
+
+# TODO : Parth - investigate doing an integration test similar to the http one. Maybe using minio or localstack
```

Comment:
(not for the merge, but for reference we have a bucket for testing and the we can get the aws keys in github, so I think we can just have an integration test against actual s3.)